internal/otg_helpers/otg_validation_helpers/otgvalidationhelpers.go

```diff
@@ -175,3 +190,34 @@ func (v *OTGValidation) ValidateECMPonLAG(t *testing.T, ate *ondatra.ATEDevice)
 
 	return nil
 }
+
+// ValidateECMP validates if equal number of packets shared across the interfaces given
+func (ev *OTGECMPValidation) ValidateECMP(t *testing.T, ate *ondatra.ATEDevice) error {
+	t.Helper()
+	totalPkts := uint64(0)
+	for _, fName := range ev.Flows {
+		totalPkts += gnmi.Get(t, ate.OTG(), gnmi.OTG().Flow(fName).Counters().InPkts().State())
+	}
+
+	var validationErrors []error
+
+	for _, p := range ev.PortWeightages {
+		actualPkts := gnmi.Get(t, ate.OTG(), gnmi.OTG().Port(ate.Port(t, p.PortName).ID()).Counters().InFrames().State())
+		expectedPkts := (float64(totalPkts) * p.Weightage) / 100
+
+		t.Logf("port: %s, Packets Received: %d", p.PortName, actualPkts)
+
+		lowerBound := expectedPkts * (1 - ev.TolerancePct)
+		upperBound := expectedPkts * (1 + ev.TolerancePct)
+
+		if float64(actualPkts) < lowerBound || float64(actualPkts) > upperBound {
+			validationErrors = append(validationErrors,
+				fmt.Errorf("port %s: actual packets %d out of expected range [%.0f - %.0f]", p.PortName, actualPkts, lowerBound, upperBound))
```

Comment:
Please return closing `)` to same indent level as opening `(`.

```suggestion
			validationErrors = append(validationErrors,
				fmt.Errorf("port %s: actual packets %d out of expected range [%.0f - %.0f]", p.PortName, actualPkts, lowerBound, upperBound),
			)
```